### DISCUSSION TITLE
Does cargo-multivers support producing static libraries?

### DISCUSSION THREAD

**Discussion by golmschenk** [ronnychevalier/cargo-multivers#12] [Q&A] [answered]
cargo-multivers doesn't support producing multiple target CPU static libraries, correct? I'm building static libraries which are linked by another (non-Rust) codebase. Ideally, I'd like to have the linker grab the appropriately optimized version of the library. Presumably, though, cargo-multivers's outer binary wrapper is the one which determines the correct CPU and there's no way to pass this information to the linker, correct? Thank you for your time.

**Comment by ronnychevalier** [comment, marked as answer]
You are correct, this is not supported.

Yes, `cargo multivers` binary wrapper ([`multivers-runner`](https://github.com/ronnychevalier/cargo-multivers/tree/830ccc0f7b85e935a0e6b260ce52a94b5cbd0a63/multivers-runner)) is the one determining at runtime the supported CPU features. Currently, the approach used by the wrapper cannot be used in the context of a library. A different approach needs to be implemented for it to work in such a context.
I'm not working on this at the moment, so unless someone is motivated to do it, it will not be supported any time soon.

**Comment by ronnychevalier** [comment]
Just in case, to make it more explicit I [added an error message](https://github.com/ronnychevalier/cargo-multivers/commit/56b509c63c3decceb99ca204133f0c65a244ff3b#diff-51c96ed71ce199893848ed7184b6f13d4b075d75a2ed63f65f80966863fd2768R284) when running `cargo multivers` on a crate that does not contain a binary.
